Modules/Core/Common/include/itkMacro.h

```diff
@@ -434,24 +434,25 @@ namespace itk
 #endif
 
 
+// Internal macro (not part of the public ITK API), used to implement `GetNameOfClass()` member functions.
+#define itkInternalGetNameOfClassImplementationMacro(thisClass)                                             \
+  {                                                                                                         \
+    static_assert(std::is_same_v<thisClass, std::remove_const_t<std::remove_reference_t<decltype(*this)>>>, \
+                  "The macro argument `" #thisClass                                                         \
+                  "` appears incorrect! It should correspond with the name of this class!");                \
+    return #thisClass;                                                                                      \
+  }                                                                                                         \
+  ITK_MACROEND_NOOP_STATEMENT
+
+
 /** Macro's used to add `GetNameOfClass()` member functions to polymorphic ITK classes: `itkVirtualGetNameOfClassMacro`
  * adds a virtual `GetNameOfClass()` member function to the class definition, and `itkOverrideGetNameOfClassMacro` adds
  * a `GetNameOfClass()` override. */
-#define itkVirtualGetNameOfClassMacro(thisClass)                                                             \
-  virtual const char * GetNameOfClass() const                                                                \
-  {                                                                                                          \
-    static_assert(std::is_same_v<thisClass, std::remove_const_t<std::remove_reference_t<decltype(*this)>>>); \
-    return #thisClass;                                                                                       \
-  }                                                                                                          \
-  ITK_MACROEND_NOOP_STATEMENT
+#define itkVirtualGetNameOfClassMacro(thisClass) \
+  virtual const char * GetNameOfClass() const itkInternalGetNameOfClassImplementationMacro(thisClass)
 
-#define itkOverrideGetNameOfClassMacro(thisClass)                                                            \
-  const char * GetNameOfClass() const override                                                               \
-  {                                                                                                          \
-    static_assert(std::is_same_v<thisClass, std::remove_const_t<std::remove_reference_t<decltype(*this)>>>); \
-    return #thisClass;                                                                                       \
-  }                                                                                                          \
-  ITK_MACROEND_NOOP_STATEMENT
+#define itkOverrideGetNameOfClassMacro(thisClass) \
+  const char * GetNameOfClass() const override itkInternalGetNameOfClassImplementationMacro(thisClass)
```

Comment:
Good idea! Should you add
```c++
#undef itkInternalGetNameOfClassImplementationMacro
```
after this line so that `itkInternalGetNameOfClassImplementationMacro` is indeed not in the public ITK API?

Reply:
Thanks for the suggestions, @SimonRit  I tried the `#undef` at the end of itkMacro.h but MSVC then gave me lots of compile errors! Apparently, the internal macro still needs to be available in the header files that actually use the GetNameOfClass macro's. So it should not be "undefined" too early. OK?

Reply:
Ok! I did not anticipate this but it makes sense. Sorry for the noise...